pipit/trace.py

```diff
@@ -127,3 +127,130 @@ def comm_matrix(self, output="size"):
             ]
 
         return communication_matrix
+
+    def match_rows(self):
+        if "Matching Index" not in self.events.columns:
+            """
+            Two columns to be added to dataframe:
+            "Matching Index" and "Matching Timestamp"
+
+            Matches dataframe indices and timestamps
+            between corresponding enter and leave rows.
+            """
+            matching_indices = [float("nan")] * len(self.events)
+            matching_times = [float("nan")] * len(self.events)
+
+            enter_leave_df = self.events.loc[
+                self.events["Event Type"].isin(["Enter", "Leave"])
+            ]
+
+            # Filter by Thread/Process
+            for id in set(enter_leave_df["Thread"]):
```

Comment:
what about filtering by process?

Reply:
Will make the necessary change to the otf2 reader and then will filter by both process and thread.
Ex: For a trace with 3 Processes and 2 Threads on each Process, the threads would now be numbered by 0 to 5, but after the change, they will be numbered 0 to 1 for each process.